src/HieDb/Create.hs

```diff
@@ -149,14 +153,20 @@ addArr (getConn -> conn) arr = do
     addName :: Name -> IO (Maybe Int64)
     addName n = case nameModule_maybe n of
       Nothing -> pure Nothing
-      Just m -> liftIO $ do
+      Just m -> do
         let occ = nameOccName n
             mod = moduleName m
             uid = moduleUnitId m
         execute conn "INSERT INTO typenames(name,mod,unit) VALUES (?,?,?)" (occ,mod,uid)
-        Just . fromOnly . head <$> query conn "SELECT id FROM typenames WHERE name = ? AND mod = ? AND unit = ?" (occ,mod,uid)
-
-addTypeRefs :: HieDb -> FilePath -> HieFile -> A.Array TypeIndex (Maybe Int64) -> IO ()
+        Just <$> lastInsertRowId conn
```

Comment:
Can't use this because the type might already be in the database, which would mean `lastInsertRowId` refers to the thing that was actually inserted.

Reply:
Ok, reverted this change.